source/extensions/matching/input_matchers/consistent_hashing/matcher.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "envoy/matcher/matcher.h"
+
+#include "common/common/hash.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace ConsistentHashing {
+
+class Matcher : public Envoy::Matcher::InputMatcher {
+public:
+  Matcher(uint32_t threshold, uint32_t modulo) : threshold_(threshold), modulo_(modulo) {}
+  bool match(absl::optional<absl::string_view> input) override {
+    // Only match if the value is present.
+    if (!input) {
+      return false;
+    }
+
+    // Otherwise, match if (hash(input) % modulo) > threshold.
+    return HashUtil::xxHash64(*input) % modulo_ > threshold_;
```

Comment:
One interesting implication of the consistent hasher API guarantee that the results are fleet-wide consistent is that if we ever change `xxHash64`, or `xxhash` itself internally makes some inconsistent breaking change, the property will be violated during rollouts. FWIW we have this problem today with our affinity load balancers. Not sure how you prefer to handle it (maybe warn in API documentation is enough?) but worth considering.

Reply:
I'll add a warning to the docs, not sure what else we could do here besides implementing our own hash that we promise we'll never change (or never update xxhash?).

Reply:
Can we model 100% match here given the `>`? Should it be `>=`?